src/bosh-dns/dns/server/handlers/forward_handler.go

```diff
@@ -116,7 +117,11 @@ func (r ForwardHandler) writeNoResponseMessage(responseWriter dns.ResponseWriter
 	case net.Error:
 		responseMessage.SetRcode(req, dns.RcodeServerFailure)
 	default:
-		responseMessage.SetRcode(req, dns.RcodeNameError)
+		if strings.Contains(err.Error(), "received SERVFAIL") {
```

Comment:
would it be possible to get the status out of the dns.Msg? The current implementation feels a bit brittle. 

Found: https://pkg.go.dev/github.com/miekg/dns#MsgHdr.String